_posts/2023-10-03-securing-wildfly-apps-oidc-openshift.adoc

```diff
@@ -0,0 +1,360 @@
+
+. Once the new user has been created, set a password for this new user from the *Credentials* tab.
+
+. From the *Role Mapping* tab, assign *alice* the *user* role. For more information, see the Keycloak documentation on how to https://www.keycloak.org/docs/latest/server_admin/index.html#proc-assigning-role-mappings_server_administration_guide[assign a role] to a user.
+
+. Create a new client as follows:
+* *General Settings*:
+** *Client type* (or *Client Protocol*, depending on your Keycloak version): *OpenID Connect*
+** *Client ID*: *myclient*
+* *Capability config*:
+** *Authentication flow*: *Standard flow*, *Direct access grants*
+* *Login settings*: Leave the fields blank for now.
+
++
+For more information, see the Keycloak documentation on how to https://www.keycloak.org/docs/latest/server_admin/index.html#_oidc_clients[Managing OpenID Connect clients].
+
+. Click *Save* to save the client. 
+
+== Add Helm Configuration
+
+. Obtain the URL for Keycloak.
++
+[source]
+----
+KEYCLOAK_URL=https://$(oc get route keycloak --template='{{ .spec.host }}') &&
+echo "" &&
+echo "Keycloak URL:   $KEYCLOAK_URL" &&
+echo ""
+----
+
+. Switch to the `charts` directory in the `simple-webapp-oidc` example.
++
+[source]
+----
+cd /PATH/TO/ELYTRON/EXAMPLES/simple-webapp-oidc/charts
+----
++
+Notice there's a `helm.yaml` file in this directory with the following content:
++
+[source]
+----
+build:
+  uri: https://github.com/wildfly-security-incubator/elytron-examples.git
+  contextDir: simple-webapp-oidc
+deploy:
+  replicas: 1
+  env:
+    - name: OIDC_PROVIDER_URL
+      value: <KEYCLOAK_URL>    <1>
+----
+<1> Replace _KEYCLOAK_URL_ with the Keycloak URL obtained in the previous command.
+
+== Deploy the Example Application to WildFly on OpenShift
+
+We can deploy our example application to WildFly on OpenShift using the WildFly Helm Chart:
+
+[source]
+----
+helm install oidc-app -f /PATH/TO/ELYTRON/EXAMPLES/simple-webapp-saml/charts/helm.yaml wildfly/wildfly
+----
+
+Notice that this command specifies the file we updated, `helm.yaml`, that contains the values
+needed to build and deploy our application.
+
+The application will now begin to build. This will take a couple of minutes.
+
+The build can be observed using:
+
+[source]
+----
+oc get build -w
+----
+
+Once complete, you can follow the deployment of the application using:
+
+[source]
+----
+oc get deployment oidc-app -w
+----
+
+Alternatively, you can check status directly from the OpenShift web console.
+
+=== Behind the Scenes
+
+While our application is building, let's take a closer look at our application.
+
+* Examine the  https://github.com/wildfly-security/elytron-examples/blob/main/simple-webapp-oidc/pom.xml[pom.xml] file.
++
+Notice that it contains an *openshift* profile.
```

Comment:
Do you think we should add a sentence about what the openshift profile means?

Reply:
Added:

"A profile in Maven lets you create a set of configuration values to customize your application build for different environments. The *openshift* profile in this example defines configurations required for OpenShift deployment."

WDYT?